src/os/bluestore/BlueFS.cc

```diff
@@ -2725,6 +2732,12 @@ int BlueFS::_flush_range(FileWriter *h, uint64_t offset, uint64_t length)
     }
   }
   vselector->add_usage(h->file->vselector_hint, h->file->fnode);
+#ifdef HAVE_LIBAIO
+  if (h->file->num_readers.load() > 0) {
```

Comment:
Shouldn't we check num_reading not num_readers instead?

Reply:
I imagine sequence of 1.read() 2.flush() 3.read().
When aio is used it is possible that (3) will read earlier then data is written to device.
So I wanted data to be synchronized if we know that some reader can do read().

It seems that you helped me uncover problem.
In sequence 1.flush 2.create reader 3.read it is possible that read will get something that is not yet flushed to disk.